credoai/reporting/dataset_base.py

```diff
@@ -0,0 +1,186 @@
+        sensitive_feature_name = list(df.drop(["count", "percentage"], axis=1).columns)[
+            0
+        ]
+
+        ax = sns.barplot(
+            x="count",
+            y=sensitive_feature_name,
+            data=df,
+            palette=plot_utils.credo_diverging_palette(1),
+            alpha=1,
+            ax=axs[0],
+        )
+        fig.patch.set_facecolor("white")
+        sns.despine()
+        ax.set_title("Data balance across " + sensitive_feature_name + " subgroups")
+        ax.set_xlabel("Number of data samples")
+        ax.set_ylabel("")
+
+        # Generate label balance barplots
+        results = results_all["balance_metrics"]["label_balance"]
+        df = pd.DataFrame(results)
+        label_name = list(df.drop([sensitive_feature_name, "count"], axis=1).columns)[0]
+
+        num_classes = df[label_name].nunique()
+        ax = sns.barplot(
+            x="count",
+            y=sensitive_feature_name,
+            hue=label_name,
+            data=df,
+            palette=plot_utils.credo_diverging_palette(num_classes),
+            alpha=1,
+            ax=axs[1],
+        )
+        fig.patch.set_facecolor("white")
+        sns.despine()
+        ax.set_title(
+            "Data balance across "
+            + sensitive_feature_name
+            + " subgroups and label values"
+        )
+        ax.set_xlabel("Number of data samples")
+        ax.set_ylabel("")
+        ax.legend(
+            bbox_to_anchor=(0.5, -0.3),
+            loc="upper center",
+            frameon=False,
+            ncol=num_classes,
+        )
+        ax.legend_.set_title(label_name)
+
+        # Generate parity metrics barplots
+        results = results_all["balance_metrics"]["metrics"]
+
+        lst = []
+        for k, v in results.items():
+            temp = pd.DataFrame(v)
+            temp["metric"] = k.replace("_", " ")
+            lst.append(temp)
+
+        df = pd.concat(lst)
+
+        ax = sns.barplot(
+            x="value",
+            y="metric",
+            hue=label_name,
+            data=df,
+            palette=plot_utils.credo_diverging_palette(num_classes),
+            alpha=1,
+            ax=axs[2],
+        )
+        fig.patch.set_facecolor("white")
+        sns.despine()
+        plt.title("Parity metrics for different preferred label value possibilities")
+        plt.xlabel("")
+        plt.ylabel("")
+        plt.legend(
+            bbox_to_anchor=(0.5, -0.2),
+            loc="upper center",
+            frameon=False,
+            ncol=num_classes,
+        )
+        ax.legend_.set_title(label_name)
+
+        plt.tight_layout()
+
+        self.figs.append(fig)
+
+        # Generate group difference barplots
+        results = results_all["group_diffs"]
+        for k, v in results.items():
+            df = pd.DataFrame(v.items(), columns=["feature", "group difference"])
+            fig = plt.figure(figsize=(8, 4), dpi=150)
+            ax = sns.barplot(
```

Comment:
Rather than create a number of different plots, create one figure with the correct number of subplots (the length of results.items()) and fill them in.

This lets us potentially create a grid of subplots, rather than just one after the other. And add a supertitle, control overal aesthetics, ec.